### PR TITLE
schedule: dox: complete api documentation

### PR DESCRIPTION
**Pull request by tlauda**
Cleans up existing documentation and adds missing one.

Signed-off-by: Tomasz Lauda <tomasz.lauda@linux.intel.com>

### CODE REVIEW

**Comment by tlauda** [comment]
@ranj063 @kv2019i Is this GLK load/unload fail a known issue? It happens pretty often in CI. Any CI issues should be prioritized.

**Comment by plbossart** [comment]
> @ranj063 @kv2019i Is this GLK load/unload fail a known issue? It happens pretty often in CI. Any CI issues should be prioritized.

yes, this popped up for a kernel rebase as well. https://github.com/thesofproject/linux/pull/2004

File: src/schedule/edf_schedule.c [hunk 1]
```diff
@@ -110,7 +110,7 @@ static void schedule_edf_task(void *data, struct task *task, uint64_t start,
 			("schedule_edf_task(), task already queued or running %d",
 			 task->state);
 		irq_local_enable(flags);
-		return;
+		return -EBUSY;
```
Comment (lgirdwood) [inline src/schedule/edf_schedule.c:113]: #define	EALREADY	114	/* Operation already in progress */ ?

**Comment by kv2019i** [comment]
@tlauda wrote:
> @ranj063 @kv2019i Is this GLK load/unload fail a known issue? It happens pretty often in CI. Any CI issues should be prioritized.

Via Pierre and Michael -> there's a bug for this https://github.com/thesofproject/sof/issues/2646

**Pull request merged**